cdk/shared_components/lambdas/check_step_function_running.py

```diff
@@ -0,0 +1,30 @@
+import boto3
+
+
+def handler(event, context):
+    state_machine_arn = event["stateMachineArn"]
+    current_execution_arn = event["currentExecutionArn"]
+
+    client = boto3.client("stepfunctions")
+
+    # List all running executions for the given state machine
+    response = client.list_executions(
+        stateMachineArn=state_machine_arn, statusFilter="RUNNING"
+    )
+    executions = response.get("executions", [])
```

Comment:
This may not matter in this context if we would only expect `current_execution_arn` to appear once but `list_executions()` is a paginated response. If `current_execution_arn` can appear multiple times in the response then we should page over the results from `list_executions` with `nextToken` to make sure `executions` is complete.